src/clzones.cpp

```diff
@@ -690,6 +691,64 @@ void zone_manager::swap( zone_data &a, zone_data &b )
     std::swap( a, b );
 }
 
+void zone_manager::start_sort( const std::vector<tripoint> &src_sorted )
+{
+    done_sorting = false;
+    to_sort.clear();
+    for( auto &src : src_sorted ) {
+        const auto &src_loc = g->m.getlocal( src );
+
+        // skip tiles in IGNORE zone and tiles on fire (to prevent taking out wood off the lit brazier)
+        // and inaccessible furniture, like filled charcoal kiln
+        if( has( zone_type_id( "LOOT_IGNORE" ), src ) ||
+            g->m.get_field( src_loc, fd_fire ) != nullptr ||
+            !g->m.can_put_items_ter_furn( src_loc ) ) {
+            continue;
+        }
+        auto items_at_src = new std::stack<item *, std::vector<item *>>();
```

Comment:
BTW this is a memory leak, these stacks are never deallocated.
The easiest option for correcting this is to just create the std::stack on the stack, and then copy it into to_sort more or less like you're doing now. Also to_sort becomes an unordered_map<tripoint, std::stack<item *, std::vector<item *>>>
Only worry about making it faster if it proves to be a problem.